pyqtgraph/graphicsItems/AxisItem.py

```diff
@@ -820,7 +820,11 @@ def tickValues(self, minVal, maxVal, size):
             ## remove any ticks that were present in higher levels
             ## we assume here that if the difference between a tick value and a previously seen tick value
             ## is less than spacing/100, then they are 'equal' and we can ignore the new tick.
-            values = list(filter(lambda x: np.all(np.abs(allValues-x) > spacing/self.scale*0.01), values))
+            close = np.any(
+                np.isclose(allValues, values[:, np.newaxis], rtol=0, atol=spacing/self.scale*0.01)
+                , axis=-1
+            )
+            values = values[~close]
             allValues = np.concatenate([allValues, values])
             ticks.append((spacing/self.scale, values))
```

Comment:
To respect the description in the method docstring, `ticks` would need to be converted to a Python list here.

Reply:
Thanks for being so attentive. I see, you had this correct already in your very first comment, but I removed it silently ;)